CMakeLists.txt

```diff
@@ -239,46 +247,57 @@ if(NOT Boost_FOUND)
 endif()
 include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
 
-target_link_libraries(OSRM ${Boost_LIBRARIES})
+# Libaries
+target_link_libraries(osrm ${Boost_LIBRARIES})
+target_link_libraries(osrm_extract ${Boost_LIBRARIES})
+target_link_libraries(osrm_contract ${Boost_LIBRARIES})
+target_link_libraries(osrm_store ${Boost_LIBRARIES})
+# Binaries
 target_link_libraries(osrm-extract ${Boost_LIBRARIES})
 target_link_libraries(osrm-prepare ${Boost_LIBRARIES})
-target_link_libraries(osrm-routed ${Boost_LIBRARIES} ${OPTIONAL_SOCKET_LIBS} OSRM)
+target_link_libraries(osrm-routed ${Boost_LIBRARIES} ${OPTIONAL_SOCKET_LIBS})
 target_link_libraries(osrm-datastore ${Boost_LIBRARIES})
+# Tests
 target_link_libraries(engine-tests ${Boost_LIBRARIES})
 target_link_libraries(extractor-tests ${Boost_LIBRARIES})
 target_link_libraries(util-tests ${Boost_LIBRARIES})
 target_link_libraries(rtree-bench ${Boost_LIBRARIES})
 
 find_package(Threads REQUIRED)
-target_link_libraries(osrm-extract ${CMAKE_THREAD_LIBS_INIT})
-target_link_libraries(osrm-datastore ${CMAKE_THREAD_LIBS_INIT})
-target_link_libraries(osrm-prepare ${CMAKE_THREAD_LIBS_INIT})
-target_link_libraries(OSRM ${CMAKE_THREAD_LIBS_INIT})
+# Libraries
+target_link_libraries(osrm ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(osrm_extract ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(osrm_contract ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(osrm_store ${CMAKE_THREAD_LIBS_INIT})
+# Tests
 target_link_libraries(engine-tests ${CMAKE_THREAD_LIBS_INIT})
 target_link_libraries(extractor-tests ${CMAKE_THREAD_LIBS_INIT})
 target_link_libraries(util-tests ${CMAKE_THREAD_LIBS_INIT})
 target_link_libraries(rtree-bench ${CMAKE_THREAD_LIBS_INIT})
 
 find_package(TBB REQUIRED)
+include_directories(SYSTEM ${TBB_INCLUDE_DIR})
 if(WIN32 AND CMAKE_BUILD_TYPE MATCHES Debug)
   set(TBB_LIBRARIES ${TBB_DEBUG_LIBRARIES})
 endif()
-target_link_libraries(osrm-datastore ${TBB_LIBRARIES})
-target_link_libraries(osrm-extract ${TBB_LIBRARIES})
-target_link_libraries(osrm-prepare ${TBB_LIBRARIES})
-target_link_libraries(osrm-routed ${TBB_LIBRARIES})
+# Libraries
+target_link_libraries(osrm_extract ${TBB_LIBRARIES})
+target_link_libraries(osrm_contract ${TBB_LIBRARIES})
+target_link_libraries(osrm_store ${TBB_LIBRARIES})
+# Tests
 target_link_libraries(engine-tests ${TBB_LIBRARIES})
 target_link_libraries(extractor-tests ${TBB_LIBRARIES})
 target_link_libraries(util-tests ${TBB_LIBRARIES})
 target_link_libraries(rtree-bench ${TBB_LIBRARIES})
-include_directories(SYSTEM ${TBB_INCLUDE_DIR})
 
 find_package( Luabind REQUIRED )
 include(check_luabind)
-
 include_directories(SYSTEM ${LUABIND_INCLUDE_DIR})
-target_link_libraries(osrm-extract ${LUABIND_LIBRARY})
-target_link_libraries(osrm-prepare ${LUABIND_LIBRARY})
+
+# Libraries
+target_link_libraries(osrm_extract ${LUABIND_LIBRARY})
+target_link_libraries(osrm_contract ${LUABIND_LIBRARY})
```

Comment:
Can we flatten the pattern we see here with `target_link_libraries`?

That is, instead of


    target_link_libraries(a ${lib_x})
    target_link_libraries(b ${lib_x})

    target_link_libraries(a ${lib_y})
    target_link_libraries(b ${lib_y})

    target_link_libraries(a ${lib_z})
    target_link_libraries(b ${lib_z})

we could just do

    target_link_libraries(a ${lib_x} ${lib_y} ${lib_z})
    target_link_libraries(b ${lib_x} ${lib_y} ${lib_z})